src/lightning/fabric/loggers/csv_logs.py

```diff
@@ -150,14 +153,15 @@ def finalize(self, status: str) -> None:
     def _get_next_version(self) -> int:
         root_dir = self.root_dir
 
-        if not os.path.isdir(root_dir):
+        if not self._fs.isdir(root_dir):
```

Comment:
I definitely don't want to block this PR as I've been asking for this fix, but should this fail on S3 which is really a key-value store and doesn't care about directories?  I think it's fine in this case but maybe it's worth a note in the docs (not sure what the appropriate place for this is) to say that the root directory must exist, even on S3.

Reply:
I don't think this is an issue because root_dir is mkdir-ed [here](https://github.com/Lightning-AI/lightning/pull/16880/files#diff-40c75290326e675f6ab4d5aa45e2dc7448b376715e294d6868bb4e1772a97eadL157), before calling `self.log_dir` which calls `_get_next_version`.
So by the time that `_get_next_version` is called, the `root_dir` already exists.

Reply:
Resolving this conversation to merge as this was a pre-existing issue. If you still think this should be changed, please open an issue and continue discussion there. Thanks!